homeassistant/components/recorder/migration.py

```diff
@@ -639,6 +639,7 @@ def _update_states_table_with_foreign_key_options(
                 _LOGGER.exception(
                     "Could not update foreign options in %s table", TABLE_STATES
                 )
+                raise
```

Comment:
I added a test to cover this line, but the test is wrong. Not sure why this is not failing in codecov.